wled00/FX.cpp

```diff
@@ -3115,6 +3115,290 @@ static uint16_t rolling_balls(void) {
 static const char _data_FX_MODE_ROLLINGBALLS[] PROGMEM = "Rolling Balls@!,# of balls,,,,Collisions,Overlay,Trails;!,!,!;!;1;m12=1"; //bar
 
 
+/*
+/  Ants (created by making modifications to the Rolling Balls code) - Bob Loeffler - January 2025
+*   bouncing balls on a track track Effect modified from Aircoookie's bouncing balls
+*   Courtesy of pjhatch (https://github.com/pjhatch)
+*   https://github.com/Aircoookie/WLED/pull/1039
+* 
+*   First slider is for the ants' speed.
+*   Second slider is for the # of ants.
+*   Third slider is for the Ants' size.
+*   
+*   Checkbox2 is for Overlay mode (enabled is Overlay, disabled is no overlay)
+*   Checkbox3 is for whether the ants will bump into each other (disabled) or just pass by each other (enabled)
+*/
+static uint16_t mode_ants(void) {
+  //allocate segment data
+  uint32_t bgcolor = BLACK;
+  static constexpr uint16_t MAX_ANTS = 32;                                     // Maximum number of ants
+  static constexpr uint8_t DEFAULT_ANT_SIZE = 1;                               
+  uint8_t antSize = DEFAULT_ANT_SIZE;
```

Comment:
Unnecessarily using `uint8_t` or `uint16_t` increases code size as compiler needs to add width checks.
Code is also slower.

If you do not need 8-bit or 16-bit numbers (for wrapping) it is best to use `int` or `unsigned`.